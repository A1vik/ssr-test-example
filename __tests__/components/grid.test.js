import React from "react";
import Grid from "../../src/shared/components/grid";
import { shallow } from "enzyme";

describe("Grid", () => {
  const props = {
      match: {
          params: {
              country: 'be',
          },
      },
  };
  test("should match snapshot if no articles", () => {
    const fetchInitialData = jest.fn().mockImplementation(() => Promise.resolve());
    global.__isBrowser__ = true;
    const wrapper = shallow(<Grid {...props} fetchInitialData={fetchInitialData} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should match snapshot with articles", () => {
      const articles = [
          {
              title: 'title',
              url: '/some/url',
              urlToImage: 'urlToImage',
          },
          {
            title: 'title2',
            url: '/some/url2',
            urlToImage: 'urlToImage2',
            publishedAt: '15-04-2018',
          },
      ];
    const fetchInitialData = jest.fn().mockImplementation(() => Promise.resolve(articles));
    global.__isBrowser__ = true;
    window.__INITIAL_DATA__ = articles;
    const wrapper = shallow(<Grid {...props} fetchInitialData={fetchInitialData} />);
    expect(wrapper).toMatchSnapshot();
  });
});
