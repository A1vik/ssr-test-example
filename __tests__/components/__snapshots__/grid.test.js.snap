// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid should match snapshot if no articles 1`] = `
<p>
  LOADING
</p>
`;

exports[`Grid should match snapshot with articles 1`] = `
<ul
  style={
    Object {
      "display": "flex",
      "flexWrap": "wrap",
    }
  }
>
  <li
    key="/some/url"
    style={
      Object {
        "margin": 30,
      }
    }
  >
    <ul>
      <li>
        <img
          alt="banner"
          src="urlToImage"
          style={
            Object {
              "height": 100,
              "width": 200,
            }
          }
        />
      </li>
      <li>
        <a
          href="/some/url"
        >
          title
        </a>
      </li>
      <li />
    </ul>
  </li>
  <li
    key="/some/url2"
    style={
      Object {
        "margin": 30,
      }
    }
  >
    <ul>
      <li>
        <img
          alt="banner"
          src="urlToImage2"
          style={
            Object {
              "height": 100,
              "width": 200,
            }
          }
        />
      </li>
      <li>
        <a
          href="/some/url2"
        >
          title2
        </a>
      </li>
      <li>
        15-04-2018
      </li>
    </ul>
  </li>
</ul>
`;
